tests/testthat/testdata/src/R/src_check_horizon_timediff.R

```diff
@@ -0,0 +1,86 @@
+                                       output_type_id_datatype = c(
+                                         "from_config", "auto", "character",
+                                         "double", "integer",
+                                         "logical", "Date"
+                                       )) {
+  checkmate::assert_class(timediff, "Period")
+  checkmate::assert_scalar(timediff)
+  checkmate::assert_character(t0_colname, len = 1L)
+  checkmate::assert_character(t1_colname, len = 1L)
+  checkmate::assert_character(horizon_colname, len = 1L)
+  checkmate::assert_choice(t0_colname, choices = names(tbl))
+  checkmate::assert_choice(t1_colname, choices = names(tbl))
+  checkmate::assert_choice(horizon_colname, choices = names(tbl))
+
+  config_tasks <- hubUtils::read_config(hub_path, "tasks")
+  output_type_id_datatype <- rlang::arg_match(output_type_id_datatype)
+  schema <- hubData::create_hub_schema(config_tasks,
+    partitions = NULL,
+    r_schema = TRUE,
+    output_type_id_datatype = output_type_id_datatype
+  )
+  assert_column_date(t0_colname, schema)
+  assert_column_date(t1_colname, schema)
+  assert_column_integer(horizon_colname, schema)
+
+  if (!lubridate::is.Date(tbl[[t0_colname]])) {
+    tbl[, t0_colname] <- as.Date(tbl[[t0_colname]])
+  }
+  if (!lubridate::is.Date(tbl[[t1_colname]])) {
+    tbl[, t1_colname] <- as.Date(tbl[[t1_colname]])
+  }
+  if (!is.integer(tbl[[horizon_colname]])) {
+    tbl[, horizon_colname] <- as.integer(tbl[[horizon_colname]])
+  }
+
+  compare <- tbl[[t0_colname]] + (timediff * tbl[[horizon_colname]]) == tbl[[t1_colname]]
+  check <- all(compare)
+  if (check) {
+    details <- NULL
+  } else {
+    invalid_vals <- paste0( # nolint: object_usage_linter
+      tbl[[t1_colname]][!compare],
+      " (horizon = ", tbl[[horizon_colname]][!compare], ")"
+    ) |> unique()
+
+    details <- cli::format_inline(
+      "t1 var value{?s} {.val {invalid_vals}} are invalid."
+    )
+  }
+
+  hubValidations::capture_check_cnd(
+    check = check,
+    file_path = file_path,
+    error = TRUE,
+    msg_subject = cli::format_inline(
+      "Time differences between t0 var {.var {t0_colname}} and t1 var
+        {.var {t1_colname}}"
+    ),
+    msg_verbs = c("all match", "do not all match"),
+    msg_attribute = cli::format_inline("expected period of {.val {timediff}} * {.var {horizon_colname}}."),
+    details = details
+  )
+}
+
+assert_column_integer <- function(colname, schema) {
+  if (schema[colname] != "integer") {
+    cli::cli_abort(
+      "Column {.arg colname} must be configured as {.cls integer} not
+      {.cls {schema[colname]}}.",
+      call = rlang::caller_call()
+    )
+  }
+}
+
+assert_column_date <- function(colname, schema) {
+  if (schema[colname] != "Date") {
+    cli::cli_abort(
+      "Column {.arg colname} must be configured as {.cls Date} not
+      {.cls {schema[colname]}}.",
+      call = rlang::caller_call()
+    )
+  }
+}
```

Comment:
**non-blocking suggestion**

The tests use this function to test for a check that uses `error = TRUE`. This function achieves that, but it does so by copying over the logic from an existing check that may change in the future.

Instead, I suggest we should make this function always return `FALSE`. This way, when the original function changes in the future, we don't have to update this one. 


```suggestion
    check <- FALSE 
  if (check) {
    details <- NULL
  } else {
    invalid_vals <- get_invalid_vals()
    details <- cli::format_inline(
      "t1 var value{?s} {.val {invalid_vals}} are invalid."
    )
  }

  hubValidations::capture_check_cnd(
    check = check,
    file_path = file_path,
    error = TRUE,
    msg_subject = cli::format_inline(
      "Time differences between t0 var {.var {t0_colname}} and t1 var
        {.var {t1_colname}}"
    ),
    msg_verbs = c("all match", "do not all match"),
    msg_attribute = cli::format_inline("expected period of {.val {timediff}} * {.var {horizon_colname}}."),
    details = details
  )
}

get_invalid_vals <- function() {
  c("2023-05-15 (horizon = 1)", "2023-05-22 (horizon = 2)")
}
```

Reply:
This is a great point! In fact I was being a bit lazy and taking the shortest path to a working testable example. I've actually refactored and simplified the materials associated with these tests to focus solely on the interaction of different conditions with custom check execution. Also added tests for early return as a result of execution errors